src/main/java/uk/gov/hmcts/reform/civil/notification/handlers/bundlecreation/BundleCreationPartiesEmailGenerator.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.hmcts.reform.civil.notification.handlers.bundlecreation;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.civil.model.CaseData;
+import uk.gov.hmcts.reform.civil.notification.handlers.EmailDTO;
+import uk.gov.hmcts.reform.civil.notification.handlers.PartiesEmailGenerator;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static uk.gov.hmcts.reform.civil.enums.MultiPartyScenario.isOneVTwoTwoLegalRep;
+
+@Slf4j
+@AllArgsConstructor
+@Component
+public class BundleCreationPartiesEmailGenerator implements PartiesEmailGenerator {
+
+    private final BundleCreationAppSolOneEmailDTOGenerator appSolOneEmailDTOGenerator;
+    private final BundleCreationRespSolOneEmailDTOGenerator respSolOneEmailDTOGenerator;
+    private final BundleCreationRespSolTwoEmailDTOGenerator respSolTwoEmailDTOGenerator;
+    private final BundleCreationClaimantEmailDTOGenerator claimantEmailDTOGenerator;
+    private final BundleCreationDefendantEmailDTOGenerator defendantEmailDTOGenerator;
+
+    @Override
+    public Set<EmailDTO> getPartiesToNotify(CaseData caseData) {
+        Set<EmailDTO> partiesToEmail = new HashSet<>();
+        log.info("Generating email for case ID: {}", caseData.getCcdCaseReference());
+        partiesToEmail.add(getApplicant(caseData));
+        partiesToEmail.addAll(getRespondents(caseData));
+
+        return partiesToEmail;
+    }
+
+    private EmailDTO getApplicant(CaseData caseData) {
```

Comment:
This can be handled with all parties email generator. Its all implemented. Do not need to override this. Refer https://github.com/hmcts/civil-service/pull/6450/files#diff-f897ca7612f87308699833efb195f6eeb90c3d239f6a92181f76ed0c3d6601c0